src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/UserPersonDetail.kt

```diff
@@ -30,33 +28,29 @@ data class UserPersonDetail(
   @JoinColumn(name = "USERNAME")
   val roles: List<UserCaseloadRole> = listOf(),
 
-  @ManyToMany(fetch = FetchType.LAZY, cascade = [CascadeType.ALL])
-  @JoinTable(name = "USER_ACCESSIBLE_CASELOADS", joinColumns = [JoinColumn(name = "USERNAME", referencedColumnName = "USERNAME")], inverseJoinColumns = [JoinColumn(name = "CASELOAD_ID", referencedColumnName = "CASELOAD_ID")])
-  val caseloads: List<Caseload> = listOf(),
+  @OneToMany(fetch = FetchType.LAZY, cascade = [CascadeType.ALL], mappedBy = "user")
+  val caseloads: List<UserCaseload> = listOf(),
```

Comment:
replaces the link table ManyToMany.
Interestingly it is not possible to keep the ManyToMany just for reads since even if the second mapping in not "updateble/insertable" JPA will delete the links regardless causing two deletes to be issued (with the second delete failing)

Reply:
interesting